test/upload-download.test.js

```diff
@@ -77,6 +78,14 @@ var ds = loopback.createDataSource({
 var Container = ds.createModel('container', {}, {base: 'Model'});
 app.model(Container);
 
+var RenamingContainer = ds.createModel('rename', {}, {base: 'Model'});
+app.model(RenamingContainer);
+
+RenamingContainer.beforeRemote('upload', function(ctx, unused, cb) {
+  ctx.targetFileName = 'renamed-test.jpg';
```

Comment:
Since the `targetFileName` is read from `options.targetFileName`, shouldn't this line set `ctx.options.targetFileName` instead?